pkg/microservice/aslan/core/common/service/kube/status.go

```diff
@@ -71,3 +72,46 @@ func GetSelectedPodsInfo(ns string, selector labels.Selector, kubeClient client.
 
 	return setting.PodRunning, ready, images
 }
+
+func GetSelectedPodsInfoWithCache(selector labels.Selector, informer informers.SharedInformerFactory, log *zap.SugaredLogger) (string, string, []string) {
+	pods, err := getter.ListPodsWithCache(selector, informer)
+	if err != nil {
+		return setting.PodError, setting.PodNotReady, nil
+	}
+
+	if len(pods) == 0 {
+		return setting.PodNonStarted, setting.PodNotReady, nil
+	}
+
+	imageSet := sets.String{}
+	for _, pod := range pods {
+		ipod := wrapper.Pod(pod)
+		imageSet.Insert(ipod.Containers()...)
+	}
+	images := imageSet.List()
+
+	ready := setting.PodReady
+
+	succeededPods := 0
+	for _, pod := range pods {
+		iPod := wrapper.Pod(pod)
+
+		// skip if pod is a succeeded job
+		if iPod.Succeeded() {
+			succeededPods++
+			continue
+		}
+
+		// 如果服务中任意一个pod不在running状态，返回unstable
+		if !iPod.Ready() {
+			return setting.PodUnstable, setting.PodNotReady, images
+		}
+	}
+
+	// return "Succeeded, Completed" if all pods are succeeded Job pods
```

Comment:
This comment can be considered for deletion, as it does not provide more understandable information than the code.